### ISSUE TITLE
Multiple partners on native slot send all native asset to DFP

### ISSUE THREAD

**Issue by haroldpussat** [prebid/Prebid.js#3769]
## Type of issue
question
## Description
Hi, 

We are currently trying to add more than 1 partner on native demand. Today, we have only Appnexus in competition. We are looking to add 2 appnexus alias in the native competition.
My slot is name "dfp_banniere_medium" and is setup with native media type like this
`banniere_medium : {native : {sponsoredBy: {sendId: true},image: {sendId: true},title: {sendId: true}, body: {sendId: true}, clickUrl: {sendId: true}}},`

There a 3 partners setup in competition (3 from Appnexus adapter, 2 with setAlias and 1 as official Alias).
Our setup if SendAllBids:false
When at least 2 partners send bids for this slot, I can see see that native value for asset are concateneted into corresponding key. So that nativeTracker can not display correctly the native style on DFP. The key concerned are all the one I setup as sendId : true

I am able to get this behavior with Prebid 2.11 (official) and our Prebid 1.39 (custom)

## Steps to reproduce
0 - SendAllBids : false, adServer DFP
1- On a full native slot, add at least 2 partners (mines are appnexus)
2 - Setup the mediaType in order to send only asset for native
3 - At least 2 partner send answer with valid bids
4 - check the targeting sent to DFP

## Test page
https://lp.additi.fr/interne/hpu/20190419-Prebid.2.11-multiple_native_partners/?dfptest=61

where there is one slot dfp_banniere_medium, setup with 3 alias appnexus partners. the native style targeted should be the one that include native asset.

### Expected results
I was expecting that only *_native_* value corresponding to the winning are sent to DFP

### Actual results
I have *_native_* value sent as many as bids received for the page, see screenshot below
![image](https://user-images.githubusercontent.com/23049515/56428351-57a1ae00-62bf-11e9-8046-e37d1e074d57.png)

The adId 72fbad640327be from bidder Appnexus366 is "well" displayed as it is the winning bid
The adId "8bc36cbffcd90c" from bidder AppnexusAst is also sent to DFP and the asset are displayed on the native style

You can see the key-value sent to DFP that include the 2 bids 👍 
![image](https://user-images.githubusercontent.com/23049515/56428469-bcf59f00-62bf-11e9-92ca-145d341bb1c2.png)

hb_native_image : hb_native_image:72fbad640327be,hb_native_image:8bc36cbffcd90c
hb_native_linkurl : hb_native_linkurl:72fbad640327be,hb_native_linkurl:8bc36cbffcd90c

## Platform details
Prebid 2.11 offical, Prebid 1.39 custom, Chrome / Firefox

**Comment by jaiminpanchal27** [comment]
@haroldpussat Thanks for reporting the issue. I think this is a bug. 

So any key which is not a standard key is merged before sending it to ad server. For e.g If Appnexus defining custom keyvalue pair foo:bar and Rubicon defining custom keyvalue pair foo:baz will be merged to foo: ['bar','baz']. In Prebid Native keys are not standard keys and hence they are merged together.

I will discuss this with the team and try to fix this asap. 

Meanwhile can you try using bidderSettings. If you use below snippet it may solve your problem. More info on [bidderSetttings here](http://prebid.org/dev-docs/publisher-api-reference.html#module_pbjs.bidderSettings)
```
pbjs.bidderSettings = {
          standard: {
              adserverTargeting: [{
                  key: "hb_bidder",
                  val: function(bidResponse) {
                      return bidResponse.bidderCode;
                  }
              }, {
                  key: "hb_adid",
                  val: function(bidResponse) {
                      return bidResponse.adId;
                  }
              }, {
                  key: "hb_pb",
                  val: function(bidResponse) {
                      return bidResponse.pbMg;
                  }
              }, {
                  key: 'hb_size',
                  val: function (bidResponse) {
                      return bidResponse.size;
                  }
              }, {
                  key: 'hb_source',
                  val: function (bidResponse) {
                      return bidResponse.source;
                  }
              }, {
                  key: 'hb_format',
                  val: function (bidResponse) {
                      return bidResponse.mediaType;
                  }
              }, {
                  key: 'hb_native_body',
                  val: function (bidResponse) {
                      return bidResponse.native.body;
                  }
              }, {
                  key: 'hb_native_linkurl',
                  val: function (bidResponse) {
                      return bidResponse.native.clickUrl;
                  }
              }, {
                  key: 'hb_native_image',
                  val: function (bidResponse) {
                      return bidResponse.native.image;
                  }
              }, {
                  key: 'hb_native_brand',
                  val: function (bidResponse) {
                      return bidResponse.native.brand;
                  }
              }, {
                  key: 'hb_native_title',
                  val: function (bidResponse) {
                      return bidResponse.native.title;
                  }
              }]
          }
      }
```

**Comment by haroldpussat** [comment]
@jaiminpanchal27 thanks for the quick feedback.
I guess your workaround seems ok when my page has only full native position.
But when I add other positions, such as standard banner, Prebid raise issue such as "cannot read property 'brand' of undefined.

![image](https://user-images.githubusercontent.com/23049515/56887188-36844e80-6a71-11e9-8a6e-42b2df891653.png)

I'm able to reproduce this issue with 1.39 custom and 2.11 official.
Please find below test page with  your workaround including other banner positions with 2.11 official 
https://lp.additi.fr/interne/hpu/20190419-Prebid.2.11-multiple_native_partners/index_bidder_settings_all_positions.html?pbjs_debug=true

Please find also test page with only native positions where your workaround is OK :
https://lp.additi.fr/interne/hpu/20190419-Prebid.2.11-multiple_native_partners/index_bidder_settings.html?pbjs_debug=true

**Comment by spormeon** [comment]
was this ever properly fixed as i get exactly the same as this, would rather use a proper fix, than a work around if possible

**Comment by spormeon** [comment]
i think you got it fixed in last release 2.29.0 as I've taken out the 'work around" and now get a single id on each bid/ targeting

**Comment by haroldpussat** [comment]
Hi, I can confirm the issue is solved with a 2.30 Prebid.js version. The only winning asset is sent to DFP. We will be able to add more than one partner in native prebid competition.
Thanks
